checkConstraints.py

```diff
@@ -28,35 +101,73 @@ def printHeader(outputFileName):
             'violated_statements'
         ]), file=outputFile)
 
-def printResults(q_id, statementCount, constraintChecks, outputFileName):
+def printResults(itemId, itemResults, outputFileName):
     with open(outputFileName, 'a') as outputFile:
-        # list of str-mapped int values, delimited by OUTPUT_DELIMITER
+        # list of str-mapped values, delimited by OUTPUT_DELIMITER
         print(OUTPUT_DELIMITER.join(map(str, [
-            q_id,
-            statementCount,
-            constraintChecks['violations'],
-            constraintChecks['warnings'],
-            constraintChecks['suggestions'],
-            constraintChecks['violated_statements']
+            itemId,
+            itemResults['statements'],
+            itemResults['violations_mandatory'],
+            itemResults['violations_normal'],
+            itemResults['violations_suggestion'],
+            itemResults['violated_statements']
             ]
         )), file=outputFile)
 
-async def countStatements(q_id):
+def logException(exception):
```

Comment:
Do we also want to print these to stderr (also for `logErrorMessage` below)? Otherwise this will very easily be overlooked. Alternatively we could add a final message to the script output, like "Warning: Some errors occurred during the last run, please check error.log".

Also error.log should probably either be emptied when the tool is re-started or a separating message indicating a new run should be there or we could (although that's not as nice) prefix each line with a timestamp.